From 2a4e6124ac1f0c7beb061365a18146ca1f1e6233 Mon Sep 17 00:00:00 2001
From: André Klitzing <aklitzing@gmail.com>
Date: Thu, 26 Mar 2020 12:52:22 +0100
Subject: Fix build with macOS 10.15 and deployment 10.12
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

io/qfilesystemengine_unix.cpp:1420:9: error: 'futimens' is only available on macOS 10.13 or newer [-Werror,-Wunguarded-availability-new]
     if (futimens(fd, ts) == -1) {
         ^~~~~~~~
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.15.sdk/usr/include/sys/stat.h:396:9: note: 'futimens' has been marked as being introduced in macOS 10.13 here, but the deployment target is macOS 10.12.0
 int     futimens(int __fd, const struct timespec __times[2]) __API_AVAILABLE(macosx(10.13), ios(11.0), tvos(11.0), watchos(4.0));
         ^
 io/qfilesystemengine_unix.cpp:1420:9: note: enclose 'futimens' in a __builtin_available check to silence this warning
     if (futimens(fd, ts) == -1) {
         ^~~~~~~~

Change-Id: Ib52adf7b1ec4f1057d8cb260a00da509429cfaed
Reviewed-by: Tor Arne Vestbø <tor.arne.vestbo@qt.io>
(cherry picked from commit 2f030c2cf3fe368be217c0e0b157e050d1c27afc)
---
 src/corelib/configure.json | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

(limited to 'src/corelib/configure.json')

diff --git a/src/corelib/configure.json b/src/corelib/configure.json
index ae360239c6..37fa85cd3d 100644
--- src/corelib/configure.json
+++ src/corelib/configure.json
@@ -344,7 +344,8 @@
                     "# Block futimens() on Apple platforms unless it's available on ALL",
                     "# deployment targets. This simplifies the logic at the call site",
                     "# dramatically, as it isn't strictly needed compared to futimes().",
-                    "darwin: QMAKE_CXXFLAGS += -Werror=unguarded-availability"
+                    "darwin: QMAKE_CXXFLAGS += -Werror=unguarded-availability -Werror=unguarded-availability-new",
+                    "CONFIG += warn_on"
                 ]
             }
         },
-- 
cgit v1.2.3-59-g8ed1b

